.github/workflows/reusable-doc-test.yaml

```diff
@@ -17,92 +17,47 @@ on:
-  #   USE_DOCKER_BUILDX_ACTION: true/false, specify whether to use the docker buildx action
-
-  # OPTIONAL organization level variables:
-  #   # Some system implementations may require different runners for building and
-  #   # running docker containers - allow specifying separately. Defaults to vars.RUNNER
-  #   # if these are not defined. Note env context nor strings are supported in the
-  #   # runs-on field, so we MUST use the vars context for the default.
-  #   RUNNER_DOCKERBUILD: runner to build docker containers
-  #   RUNNER_DOCKERRUN: runner to run docker containers
-
   # The path to the GeoIPS package in DOCLINTTEST images
   BASE_GEOIPS_PATH: /packages/geoips
-  # Allow overriding the doclinttest image tag used for plugin packages
-  # Allows using a different version of the doc, lint, and test code
-  PLUGIN_DOCLINTTEST_TAG: doclinttest-stable
 
 defaults:
   run:
     shell: bash
 
 jobs:
+  # Build the base GeoIPS doclinttest image
+  # This image is used in all subsequent tests
+  # It is only built for the main GeoIPS package but is used by all repositories
+  #
+  # Within this image, the GeoIPS package is located in /packages/geoips and is
+  # installed using `pip install .[doc,lint,test]`. Scripts from geoips can be called
+  # from /packages/geoips.
+  get-doclinttest-image:
+    name: Get docker image
+    # You do not appear to be able to use variables in this "uses" field.
+    uses: NRLMMD-GEOIPS/geoips_ci/.github/workflows/reusable-build-docker-image.yaml@build_stable_image_on_tag
+    secrets:
+      token: ${{ secrets.token }}
+      docker_token: ${{ secrets.docker_token }}
+    permissions:
+      packages: write
+      contents: write
+
   # Set variables that should be accessible to all jobs
   get-job-flags:
     runs-on: ${{ vars.RUNNER }}
+    needs: [get-doclinttest-image]
     env:
       CURR_REPO: ${{ github.event.repository.name }}
     outputs:
-      # The name of the untagged docker images to be created by this workflow
-      IMAGE_NAME: ${{ steps.set_image_name.outputs.IMAGE_NAME }}
-      REPO_NAME: ${{ steps.set_image_name.outputs.REPO_NAME }}
-      # The tag to use for the doclinttest images
-      DOCLINTTEST_TAG: ${{ steps.set_doclinttest_tag.outputs.DOCLINTTEST_TAG }}
-      # The image to use for doclinttest jobs
-      DOCLINTTEST_IMAGE: ${{ steps.set_doclinttest_image.outputs.DOCLINTTEST_IMAGE }}
       # Flags for enabling/disabling various parts of the workflow
       # Default to enabled unless overridden by a repository-level variable
       CI_BUILD_SPHINX_HTML: ${{ steps.get_job_flags.outputs.CI_BUILD_SPHINX_HTML }}
       CI_BUILD_SPHINX_PDF: ${{ steps.get_job_flags.outputs.CI_BUILD_SPHINX_PDF }}
       CI_TEST_INTERFACES: ${{ steps.get_job_flags.outputs.CI_TEST_INTERFACES }}
       CI_TEST_PYTEST_SHORT: ${{ steps.get_job_flags.outputs.CI_TEST_PYTEST_SHORT }}
+      DOCKER_IMAGE_NAME: ${{ needs.get-doclinttest-image.outputs.DOCKER_IMAGE_NAME }}
```

Comment:
The "DOCKER_IMAGE_NAME" output is defined at the beginning of reusable-build-docker-image.yaml, and we are grabbing it here to set in our local "get-job-flags"